Add function whose return data from .tcx files

nero2310 opened the pull request:
This function return data as Datatable object list, aslo return
Dictonary with exceptions

nero2310 committed:
Add function whose return data from .tcx files

This function return data as Datatable object list, aslo return
Dictonary with exceptions

mdyzma commented on hdp/process/load.py:
```diff
@@ -183,9 +186,36 @@ def load_gpx(gpx_files: list, params: dict):
 
 
 # Team 4
 def load_jpg(jpg_files: list, params: dict):
```
This function ddoes not do anytjhing Use YAGNI (you are not gonna need it). Please, do not place code without function.

I suggest use Project tab and create issue/bug.

nero2310 replied:
Do you mean this load_jpg function? I am pretty sure this is empty for a purpose.

mdyzma commented on hdp/process/load.py:
```diff
@@ -183,9 +186,36 @@ def load_gpx(gpx_files: list, params: dict):
+                            if isinstance(value,NavigableString):
+                                point[value.name] = value.string
+                            elif isinstance(value,Tag):
+                                point[value.name] = value.text.strip()
```
Too complicated/nested function, try to extract to higher level function.

i.e.:

`def extract_path():
    pass`


```python
def extract_all_trackpoints(node_name="trackpoint"):
    for trackpoint in activity.find_all(node_name):
        points = {element.name: element.text for element in trackpoint.children if
             isinstance(element, Tag) and len(element.findChildren()) == 0}
    return points

def extract_points(trackpoints):
    point = {}
    for element in trackpoints.children:
        if isinstance(element,Tag) and len(element.findChildren())>1:
            for value in element.children:
                if isinstance(value,NavigableString):
                    point[value.name] = value.string
                elif isinstance(value,Tag):
                    point[value.name] = value.text.strip()
    return point
```

If it still complains refactor further.

nero2310 replied:
Should I define auxiliary functions inside load_tcx or outside?

nero2310 committed:
Use black to format file

mdyzma commented on hdp/process/load.py:
```diff
@@ -1,4 +1,7 @@
 import pandas as pd
+from bs4 import BeautifulSoup
+from bs4 import NavigableString, Tag
+
```
Import proper typing suggestions
```suggestion
from typing import List, Tuple, Dict
```

mdyzma commented on hdp/process/load.py:
```diff
@@ -183,9 +186,36 @@ def load_gpx(gpx_files: list, params: dict):
 
 # Team 3
 def load_tcx(tcx_files: list, params: dict):
```
```suggestion
def load_tcx(tcx_files: List[str], params: Dict) -> Tuple:
```

nero2310 committed:
Add better type hints, remove params from load_tcx.

BugFix fix a problem that casued the insertion of excessive trackpoint
copies

nero2310 committed:
Add default value for function parameters.

Is no longer for user to pass empty dict in function calling is user
don't want pass parameters

mdyzma commented:
Outside

czw., 2 lip 2020, 16:30 użytkownik nero2310 <notifications@github.com>
napisał:

> *@nero2310* commented on this pull request.
> ------------------------------
>
> In hdp/process/load.py
> <https://github.com/Open-Health-Data-Project/hdp/pull/4#discussion_r449043964>
> :
>
> > +    activites = []
> +    for path in tcx_files:
> +        try:
> +            name = _get_file_name(path)
> +            with(open(path,"r")) as file:
> +                activity = BeautifulSoup(file,'lxml')
> +            for trackpoint in activity.find_all("trackpoint"):
> +                point = {element.name: element.text for element in trackpoint.children if
> +                         isinstance(element, Tag) and len(element.findChildren()) == 0}
> +                for element in trackpoint.children:
> +                    if isinstance(element,Tag) and len(element.findChildren())>1:
> +                        for value in element.children:
> +                            if isinstance(value,NavigableString):
> +                                point[value.name] = value.string
> +                            elif isinstance(value,Tag):
> +                                point[value.name] = value.text.strip()
>
> Should I define auxiliary functions inside load_tcx or outside?
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/Open-Health-Data-Project/hdp/pull/4#discussion_r449043964>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABV5Z5EMSATG5GBGZFZN3ZTRZSKWTANCNFSM4ONPI4YA>
> .
>

nero2310 committed:
Refactoring function whose read data from tcx file

Optimaize imports, add documentation to function

nero2310 committed:
Remove blank line from file end
